app/display/representation-javafx/src/main/java/org/csstudio/display/builder/representation/javafx/widgets/EmbeddedDisplayRepresentation.java

```diff
@@ -376,40 +380,42 @@ public void updateChanges()
         {
             final Integer width = model_widget.propWidth().getValue();
             final Integer height = model_widget.propHeight().getValue();
-            scroll.setPrefSize(width, height);
+
+            // zoom also applies to width and height so we have to de-scale them
+            final Double scaled_width = width / zoom_factor_x;
+            final Double scaled_height = height / zoom_factor_y;
+
+            // update minimum size with zoom factor applied 'in reverse'
+            inner.setMinSize(scaled_width, scaled_height);
+
+            // set minimum and maximum size of jfx_node
+            // to match the requested size
+            jfx_node.setMinSize(width, height);
+            jfx_node.setMaxSize(width, height);
 
             final Resize resize = model_widget.propResize().getValue();
 
-            // Does jfx_node need to crop,
-            // or will resizing resp. scroll pane handle it?
-            if (resize == Resize.Crop)
-                jfx_node.setClip(new Rectangle(width, height));
-            else
-                jfx_node.setClip(null);
+            zoom.setX(zoom_factor_x);
+            zoom.setY(zoom_factor_y);
 
             if (resize == Resize.None)
             {
                 // Need a scroll pane (which disables itself as needed)
                 jfx_node.getChildren().setAll(scroll);
+                scroll.setPrefSize(width, height);
+                inner.setClip(null);
                 scroll.setContent(inner);
-
-                zoom.setX(1.0);
-                zoom.setY(1.0);
             }
             else
             {   // Don't use a scroll pane
+                scroll.setContent(null);
                 jfx_node.getChildren().setAll(inner);
 
-                if (resize == Resize.ResizeContent  ||  resize == Resize.StretchContent )
-                {
-                    zoom.setX(zoom_factor_x);
-                    zoom.setY(zoom_factor_y);
-                }
-                else // Resize.Crop, SizeToContent
-                {
-                    zoom.setX(1.0);
-                    zoom.setY(1.0);
-                }
+                // During runtime or if the resize property is set to Crop we clip inner
+                // but allow 'overdrawing' in edit mode so the out-of-region widgets are visible to the user

```

Comment:
Allow 'overdrawing' in edit mode:
On one hand, that sounds useful.
On the other, it means the display looks different in edit mode, showing content outside of the bounds which will then vanish at runtime. Could add another 'outline' border in edit mode to indicate the container size, making it very obvious what's outside and about to be clipped at runtime. But then adding such a border means yet another difference between edit and runtime ..
Let's go with what you have for now; we can tweak the 'overdrawing', additional outline etc. later if it really turns into an issue.

Reply:
You are right; it is not "nice" to render the content differently during runtime but this way it will be obvious what is causing it when someone opens an issue. Plus it is also a kind of reminder that this needs to be addressed eventually.
The additional border is a good idea but I don't know how to get what is outside `inner`; what is "outside" of the embedded display in the parent display is also "outside" in the child. Maybe we should render a semi transparent rectangle filled with the color "INVALID" over everything that is outside the limits of its display?